content/blog/2022-07-28-switching-to-dvc-from-jupyter-vscode.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Converting a Jupyter Notebook to a DVC Project
+date: 2022-07-28
+description: >
+  Working with notebooks is common in machine learning. That's why we're
+  covering some tools that make it easy to do more with a complex project.
+descriptionLong: >
+  Once you've run some experiments in a Jupyter notebook, you know that you
+  can't save each experiment. Now, if you're using the Jupyter VSCode extension,
+  we can show you how to make those experiments reproducible with the addition
+  of the DVC VSCode extension.
+picture: 2022-07-28/jupyter-to-dvc.png
+pictureComment: Using the DVC VSCode Extension with a Jupyter Notebook
+author: milecia_mcgregor
+commentsUrl: https://discuss.dvc.org/t/syncing-data-to-aws-s3/1192
+tags:
+  - MLOps
+  - DVC
+  - Git
+  - VSCode
+  - Juptyer Notebooks
+---
+
+For many machine learning engineers, the starting point of a project is a
+Jupyter notebook. This is fine for running a few experiments, but there comes a
+point where you need to scale the project to accomodate hundreds or even
+thousands more experiments. It will also be important to track the experiments
+you run so that when you find an exceptional model, you'll be able to reproduce
+it and get it ready for production.
+
+In this tutorial, we're going to start a project with a Juptyer notebook in
+VSCode. Then we'll convert it to DVC to make reproducible experiments and use
+the DVC VSCode extension to run new experiments and see how to compare them all.
+[Here's the project](https://github.com/iterative/stale-model-example/tree/jupyter-to-dvc)
+we'll be working with.
+
+## Start training with the notebook
+
+Many times you'll start an ML project with a few cells in a notebook just to
+test out some thoughts you might have. So you might have a simple notebook where
+you set some hyperparameters, load your data, train a model and evaluate its
+metrics, and then save the model. That's what we're doing in the
+`bicycle_experiments.ipynb` file.
+
+![Jupyter notebook cells](/uploads/images/2022-07-28/jupyter-notebook.png)
```

Comment:
Does this need to be a screenshot, or could it be a code snippet?

Reply:
I can do a code snippet. I wasn't quite sure if it was better to visually show the cells or show the code.

Reply:
Sorry, but I don't understand why we would put a code snippet when the purpose of the post is to go from a Jupyter Notebook to the yaml files and this paragraph is referring to a Notebook.  Can someone explain to me what I'm missing?

Reply:
I think there was confusion here, and I assume @RCdeWit did not intend to show the underlying `.ipynb` file contents! It would be great if we could show the notebook but still be able to copy and paste the code, but I guess maybe a screen shot is best unless there is a way to embed the notebook or some portion of it here.

Reply:
Ok, cool. I also got confused here. 😅 I put the screenshot back, but I don't think there's a way to embed the notebook here.